pi-top/pi-topOS-Core-Packages
# Set hostname based on pi-top serial

Use the pi-top serial code to set the device `hostname` through the `pt-hostname` script (which is installed by the new `pt-os-hostname` package). Based on the argument provided to the script, it will:

1. Modify the device hostname on boot if necessary. 

This runs every time on boot since a user can swap SD cards from one pi-top to another; since the hostname is now based on the pi-top, the hostname will still identify a particular pi-top (well, as long as the hub wasn't changed). 

This change means that the device is now accessible via `pi-top-XXXX.local` instead of the old `pi-top.local`, since the service `avahi-daemon`, which is in charge of publishing the hostname in a network, uses the system hostname.

2. Publish `pi-top.local` as an alias for the device. This means that the device can still be accessed through `pi-top.local`.

The is done by the `pt-network-alias` service which will basically run `avahi-publish`, making `pi-top.local` an alias of whatever the IP address of the network interface `ptusb0` is.

### How to test? 

- Download the deb files from the PR artifacts. This repo creates multiple packages; the only package we need to install is `pt-os-hostname` (e.g.: `pt-os-hostname_5.3.0-3~15.gbp5a2ca0_all.deb`).
- scp the deb for `pt-os-hostname` into your pi-top
- Make sure your pi-top has an internet connection since it will need to install a new dependency, `avahi-utils`.
- ssh into your pi-top and install the debs using `sudo apt install ./*.deb`
- Reboot your device
- ssh again and make sure of the following:
1. The prompt of the terminal uses the new hostname. eg: `pi@pi-top-E89E $` 
2. Verify output of the following commands to confirm the new hostname is used in the OS:
```
$ cat /etc/hostname
pi-top-E89E

$ cat /proc/sys/kernel/hostname
pi-top-E89E

$ hostname
pi-top-E89E

$ hostnamectl status
   Static hostname: pi-top-E89E
         Icon name: computer
        Machine ID: 09b678629d2c4b4cbbb354b4bbe380ad
           Boot ID: 5646754706a14474a4597bbd12911aba
  Operating System: Raspbian GNU/Linux 11 (bullseye)
            Kernel: Linux 5.10.63-v7l+
      Architecture: arm

$ cat /etc/hosts
127.0.0.1   localhost
::1      localhost ip6-localhost ip6-loopback
ff02::1     ip6-allnodes
ff02::2     ip6-allrouters

127.0.1.1   pi-top-E89E
```
3. From your computer, try to ping the pi-top using `pi-top.local` and the new serial number based hostname:
```
$ ping -c1 pi-top-E89E.local
PING pi-top-e89e.local (192.168.64.1): 56 data bytes
64 bytes from 192.168.64.1: icmp_seq=0 ttl=64 time=0.686 ms

$ ping -c1 pi-top.local
PING pi-top.local (192.168.64.1): 56 data bytes
64 bytes from 192.168.64.1: icmp_seq=0 ttl=64 time=0.532 ms
```
